File: 2026/05-May/20-WEDNESDAY/solution.py
class Solution:
    def findThePrefixCommonArray(self, A, B):
        n = len(A)
        count = [0]*(n+1)
        ans = []
        common = 0

        for a, b in zip(A, B):
            count[a] += 1
            if count[a] == 2:
                common += 1

            count[b] += 1
            if count[b] == 2:
                common += 1

            ans.append(common)

        return ans

File: 2026/05-May/20-WEDNESDAY/solution.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        int n = A.size();
        vector<int> ans;
        vector<int> count(n + 1, 0);

        int common = 0;

        for(int i = 0; i < n; i++) {
            if(++count[A[i]] == 2)
                common++;

            if(++count[B[i]] == 2)
                common++;

            ans.push_back(common);
        }

        return ans;
    }
};
